network/p2p/inspector/validation/errors.go

```diff
@@ -51,23 +52,45 @@ func IsIWantCacheMissThresholdErr(err error) bool {
 	return errors.As(err, &e)
 }
 
-// DuplicateFoundErr error that indicates a duplicate has been detected. This can be duplicate topic or message ID tracking.
-type DuplicateFoundErr struct {
-	err error
+// DuplicateTopicErr error that indicates a duplicate has been detected. This can be duplicate topic or message ID tracking.
+type DuplicateTopicErr struct {
+	topic   string
+	msgType p2pmsg.ControlMessageType
 }
 
-func (e DuplicateFoundErr) Error() string {
-	return e.err.Error()
+func (e DuplicateTopicErr) Error() string {
+	return fmt.Sprintf("duplicate topic foud in %s control message type: %s", e.msgType, e.topic)
+}
+
+// NewDuplicateTopicErr returns a new DuplicateTopicErr.
+func NewDuplicateTopicErr(topic string, msgType p2pmsg.ControlMessageType) DuplicateTopicErr {
+	return DuplicateTopicErr{topic, msgType}
+}
+
+// IsDuplicateTopicErr returns true if an error is DuplicateTopicErr.
+func IsDuplicateTopicErr(err error) bool {
```

Comment:
This function is only used by other tests, so it's more of a test helper. Perhaps it should be moved to `errors_test.go` to indicate that it's only meant to be used by tests?

Reply:
Technically these methods could potentially be used outside the test, also this is the pattern we are using throughout the code base. It may cause some confusion that the errors and these methods are defined in separate places lets keep them together.